actions/generate_changelog.go

```diff
@@ -0,0 +1,93 @@
+package actions
+
+import (
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/codegangsta/cli"
+	"github.com/google/go-github/github"
+)
+
+const (
+	changelogTpl string = `{{.OldRelease}} -> {{.NewRelease}}
+
+# Features
+
+{{range .Features}} - {{.}}
+{{else}}No new features for this release.
+{{end}}
+
+# Fixes
+
+{{range .Fixes}} - {{.}}
+{{else}}No bug fixes for this release.
+{{end}}
+
+# Documentation
+
+{{range .Documentation}} - {{.}}
+{{else}}No new documentation for this release.
+{{end}}
+
+# Maintenance
+
+{{range .Maintenance}} - {{.}}
+{{else}}No maintenance required for this release.
+{{end}}`
+)
+
+type Changelog struct {
+	OldRelease string
+	NewRelease string
+	Features []string
+	Fixes []string
+	Documentation []string
+	Maintenance []string
+}
+
+// GenerateChangelog is the CLI action for creating an aggregated changelog from all of the Deis Workflow repos.
+func GenerateChangelog(ghClient *github.Client) func(*cli.Context) {
+	return func(c *cli.Context) {
+		changelog := &Changelog{
+			OldRelease: c.Args().Get(0),
+			NewRelease: c.Args().Get(1),
+		}
+		if changelog.OldRelease == "" || changelog.NewRelease == "" {
+			log.Fatal("Usage: generate-changelog <old-release> <new-release>")
+		}
+		for _, name := range repoNames {
+			commitCompare, resp, err := ghClient.Repositories.CompareCommits("deis", name, changelog.OldRelease, changelog.NewRelease)
```

Comment:
see line 61. The whole function aggregates changelog entries from each repo in serial. I'm going to re-write this to do it in parallel and add tests as well.